Sources/PackageCollections/Utility.swift

```diff
@@ -0,0 +1,74 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import class Foundation.FileManager
+import TSCBasic
+
+import PackageModel
+import SourceControl
+
+// FIXME: Need a better place to put this + maybe better name
```

Comment:
👀

Reply:
This probably belongs down in TSC.  There are similar methods scattered there, such as for finding the temporary directory on the platform, etc.

Reply:
the issue is that this is SwiftPM specific. we can move to TSC and make it take an argument, but that kind defeats the purpose?

Reply:
Yes, I can see that point.  Another way to do that would be to have a facility in TSC where a client could register the specifics, but for anything like that, we should probably do it as a separate PR.

Reply:
@abertelrud were do you think this should move to? I can see this useful in the context of #2985 as well

Reply:
That's a bit tricky.  One option is `ProcessEnv`, whose name sounds as if it just deals with environment variables but which actually has current working directory and seems intended to also support other environmental state.  There would be a generic cache directory whose actual location would depend on the platform, and that a client such as SwiftPM could customize.

If we think about this as more of a specialized directory it would probably instead be some registry in TSCUtility where clients can register paths with well-known identifiers, and one of those would be for the cache location.  I don't want to make this too complicated so I think maybe just a way to get at the default cache directory for the process in `ProcessEnv`, which would default to `FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)` or fall back on `~/.swiftpm/cache`.

Those two options presuppose that the cache can be in the local file system.  If we really want it to be in a particular file system, then I'd probably add this as an extension on FileSystem (perhaps in TSCUtility).

I know that's not a particularly definitive opinion, but I think it depends on whether we really want this cache to be in a particular file system or just the process's generic system cache.

Reply:
Actually: is it even feasible to use a cache location in an arbitrary file system for this particular case?  I'm guessing that sqlite requires a real file system.

Reply:
Ok, so in response to my own question at the end there, I see now that the SQLite wrapper supports in-memory databases, which explains why InMemoryFileSystem is even an option to use here.

Back to the actual question here, I think there should either be an addition to ProcessEnv to return the path of a cache directory that has a default value based on the platform but that can be customized by the client process.  I don't know if we need to support a registry of such paths, or if a single cache directory accessor would make sense to start with.